src/jtag3.c

```diff
@@ -1585,18 +1585,18 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
           int sscanf_success = sscanf(extended_param, "vtarg=%lf", &vtarg_set_val);
           PDATA(pgm)->vtarg_data = (double)((int)(vtarg_set_val * 100 + .5)) / 100;
           if (sscanf_success < 1 || vtarg_set_val < 0) {
-            pmsg_error("invalid vtarg value '%s'\n", extended_param);
+            pmsg_error("invalid vtarg value %s\n", extended_param);
             rv = -1;
             break;
           }
           PDATA(pgm)->vtarg_set = true;
-          continue;
         }
         // Get target voltage
-        else if(str_eq(extended_param, "vtarg")) {
+        else if(str_eq(extended_param, "vtarg"))
           PDATA(pgm)->vtarg_get = true;
-          continue;
-        }
+        else
```

Comment:
I am not convinced about this flow of control. Should further -x... processing be stopped through this `break` and `rv` returned (whatever it might be at this point?) 

Generally, I find this parsing function has grown over time (not the fault of this PR) into something that is hard to read (but this PR is the straw that breaks the camel's back).

One thing that would make it less hard to read is to remove the `else` after `continue` or `break`.

Another thing: When the code `break`s out of the loop it should set `rv = -1` before and print an error message. `break` should only be used for errors (as it leaves potentially other options unparsed)

Both measures would make this *a lot* more readable

Reply:
I've pushed a commit that solves most (all?) of the issues you pointed out. Please review.

Reply:
Much better! Thanks for carefully compartmentalising the parsing options. Two small remaining comments (see code comments)